data/sql/updates/pending_db_world/rev_1605128047204479800.sql

```diff
@@ -0,0 +1,147 @@
+    -- Alliance quests only
+    SELECT `id`,
+            1 AS requirement_type,
+            `quest_done_A`,
+            0 AS faction,
+			quest_failed_text
+    FROM `access_requirement`
+    WHERE `quest_done_A` > 0 AND `quest_done_A` != `quest_done_H`
+    UNION
+    -- Horde quests only
+    SELECT `id`,
+            1 AS requirement_type,
+            `quest_done_H`,
+            1 AS faction,
+			quest_failed_text
+    FROM `access_requirement`
+    WHERE `quest_done_H` > 0 AND `quest_done_A` != `quest_done_H`
+    UNION
+    -- Both factions
+    SELECT `id`,
+            1 AS requirement_type,
+            `quest_done_H`,
+            2 AS faction,
+			quest_failed_text
+    FROM `access_requirement`
+    WHERE `quest_done_H` > 0 AND `quest_done_A` = `quest_done_H`
+;
+
+-- REORDER PRIMARY KEY
+CREATE TEMPORARY TABLE IF NOT EXISTS `temp` select * from `dungeon_access_requirements` ORDER BY dungeon_access_id, faction;
+TRUNCATE `dungeon_access_requirements`;
+ALTER TABLE `dungeon_access_requirements`; 
+INSERT INTO `dungeon_access_requirements` SELECT * FROM `temp`;
+
+-- Verify result with this
+-- SELECT `id`, `quest_failed_text` FROM `access_requirement` WHERE `quest_failed_text` IS NOT NULL;
+
+
+-- Remove columns + rename table
+ALTER TABLE `access_requirement`
+DROP `item`,
+DROP `item2`,
+DROP `quest_done_A`,
+DROP `quest_done_H`,
+DROP `completed_achievement`,
+DROP `quest_failed_text`,
+RENAME TO `dungeon_access_template`,
+COMMENT='Dungeon/raid access template and single requirements';
+
+-- Rename columns
+ALTER TABLE `dungeon_access_template`
+CHANGE `mapId` `map_id` mediumint unsigned NOT NULL COMMENT 'Map ID from instance_template' AFTER `id`,
+CHANGE `difficulty` `difficulty` tinyint unsigned NOT NULL DEFAULT 0 COMMENT '5 man: 0 = normal, 1 = heroic, 2 = epic (not implemented) | 10 man: 0 = normal, 2 = heroic | 25 man: 1 = normal, 3 = heroic' AFTER `map_id`,
+CHANGE `level_min` `min_level` tinyint unsigned NULL DEFAULT NULL AFTER `difficulty`,
+CHANGE `level_max` `max_level` tinyint unsigned NULL DEFAULT NULL AFTER `min_level`,
+CHANGE `item_level` `min_avg_item_level` smallint unsigned NULL DEFAULT NULL COMMENT 'Min average ilvl required to enter' AFTER `max_level`,
+CHANGE `comment` `comment` varchar(255) COLLATE 'utf8_general_ci' NULL COMMENT 'Dungeon Name 5/10/25/40 man - Normal/Heroic' AFTER `min_avg_item_level`;
+
+-- Add KEY CONSTRAINTS
+ALTER TABLE `dungeon_access_template` ADD CONSTRAINT `FK_dungeon_access_template__instance_template` FOREIGN KEY (`map_id`) REFERENCES `instance_template` (`map`) ON DELETE CASCADE ON UPDATE CASCADE;
+ALTER TABLE `dungeon_access_requirements` ADD CONSTRAINT `FK_dungeon_access_requirements__dungeon_access_template` FOREIGN KEY (`dungeon_access_id`) REFERENCES `dungeon_access_template` (`id`) ON DELETE CASCADE ON UPDATE CASCADE;
+
+
+-- Add the acore_strings
+DELETE FROM `acore_string` WHERE `entry` IN (882,883,884,885,886,887,888);
```

Comment:
@Kitzunu you told him to DELETE these but guess what? some of us had used one of these entries, and it would destroy this without warning and it's almost impossible to be aware of this unless you retest every of your custom features

so I'm in favour of removing the "DELETE FROM", that's how i discovered some of my code (not made by me) was using one of these strings

Reply:
But then you cant run it twice and thereby breaking you can breal the import.

Im a firm believer that Core goes before custom.

Reply:
how do you run it twice? how do you import the SQL?

Reply:
I agree with @Kitzunu , the DB updates should be resilient enough to not fail just because there are some custom contents using the same IDs. I can see @BarbzYHOOL point, but I think any custom SQL should be kept in a separate file and having IDs/GUIDs paremetrised into variables so you can easily re-apply them on top of a latest AC, changing the IDs/GUIDs values when needed. So if you notice something wrong about your custom contents, you can just re-apply your SQL contents with minimal effort. For example: https://github.com/azerothcore/portals-in-all-capitals

Reply:
you can't notice something wrong as i said... there are 1200 strings in acore_string, imagine you have several custom changes, let's say you have 10 custom modules and each uses 5 acore_string

you would have to test every case where these 5 strings occur EVERYTIME someone sneakily adds new acore_string into the core, this is unmanageable

however while writing this I feel like the issue comes from using IDs that are non used, so it is a mistake of the custom developer and solely his mistake (and because we don't have a robust solution for custom IDs). So in the end, I guess it's fine, it will teach devs to be careful I guess :(